### DISCUSSION TITLE
How to obtain ETF data?

### DISCUSSION THREAD

**Discussion by yuselg** [alvarobartt/investpy#399] [Q&A] [answered]
(1) VIG is a US ETF available at investing.com. But the following error is displayed: ERR#0034: country " + country.lower() + " not found
df = investpy.get_stock_historical_data(stock='VIG',
                                        country='United States',
                                        from_date='01/01/2010',
                                        to_date='01/01/2020')

(2) Likewise HCON is a Canadian ETF, with a similar error.
df = investpy.get_stock_historical_data(stock='HCON',
                                        country='canada',
                                         from_date='01/01/2010',
                                         to_date='13/06/2021')

Is there a special way to request ETF data? Thanks

**Comment by alvarobartt** [comment, marked as answer]
Hi @yuselg this is weird... :confounded: I'll have a look at it later today so as to provide you a more detailed answer as that functions should work as long as the data is available in the static CSV files of `investpy`, but anyway, just as a fix/workaround you can use `investpy.search_quotes` function which uses the Investing.com search engine :+1:

Here's a usage example:

```python
import investpy
search_result = investpy.search_quotes(text='VIG', products=['etfs'], countries=['united states'], n_results=1)
print(search_result)

recent_data = search_result.retrieve_recent_data()
historical_data = search_result.retrieve_historical_data(from_date='01/01/2019', to_date='01/01/2020')
information = search_result.retrieve_information()
default_currency = search_result.retrieve_currency()
technical_indicators = search_result.retrieve_technical_indicators(interval='daily')
```

For more information please visit the documentation of `investpy.search_quotes` at https://investpy.readthedocs.io/_info/usage.html#search-live-data

**Reply by yuselg** [reply]
Thank you, that works!

**Reply by cla-azzarello** [reply]
Hi, how can I have a JSON output by search_result.retrieve_historical_data() ?

Thanks in advance.
